subprojects/jacoco/src/integTest/groovy/org/gradle/testing/jacoco/plugins/JacocoAggregationIntegrationTest.groovy

```diff
@@ -0,0 +1,486 @@
+        file("application/build/reports/jacoco/integTestCodeCoverageReport/html/index.html").assertExists()
+
+        def testReport = new JacocoReportXmlFixture(file("application/build/reports/jacoco/testCodeCoverageReport/testCodeCoverageReport.xml"))
+        testReport.assertHasClassCoverage("application.Adder")
+        testReport.assertHasClassCoverage("direct.Multiplier")
+        testReport.assertHasClassCoverage("transitive.Powerize")
+        testReport.assertHasClassCoverage("transitive.Divisor", 0)
+
+        def integTestReport = new JacocoReportXmlFixture(file("application/build/reports/jacoco/integTestCodeCoverageReport/integTestCodeCoverageReport.xml"))
+        integTestReport.assertHasClassCoverage("application.Adder", 0)
+        integTestReport.assertHasClassCoverage("direct.Multiplier", 0)
+        integTestReport.assertHasClassCoverage("transitive.Powerize", 0)
+        integTestReport.assertHasClassCoverage("transitive.Divisor")
+    }
+
+    def "can aggregate jacoco reports from root project"() {
+        given:
+        buildFile << """
+            apply plugin: 'org.gradle.jacoco-report-aggregation'
+
+            dependencies {
+                jacocoAggregation project(":application")
+                jacocoAggregation project(":direct")
+            }
+
+            reporting {
+                reports {
+                    testCodeCoverageReport(JacocoCoverageReport) {
+                        testType = TestType.UNIT_TESTS
+                    }
+                }
+            }
+        """
+
+        when:
+        succeeds(":testCodeCoverageReport")
+
+        then:
+        file("transitive/build/jacoco/test.exec").assertExists()
+        file("direct/build/jacoco/test.exec").assertExists()
+        file("application/build/jacoco/test.exec").assertExists()
+
+        file("build/reports/jacoco/testCodeCoverageReport/html/index.html").assertExists()
+
+        def report = new JacocoReportXmlFixture(file("build/reports/jacoco/testCodeCoverageReport/testCodeCoverageReport.xml"))
+        report.assertHasClassCoverage("application.Adder")
+        report.assertHasClassCoverage("direct.Multiplier")
+        report.assertHasClassCoverage("transitive.Powerize")
+    }
+
+    def 'test verification failure prevents creation of aggregated report'() {
+        given:
+        file("application/build.gradle") << """
+            apply plugin: 'org.gradle.jacoco-report-aggregation'
+        """
+        file("direct/src/test/java/direct/MultiplierTest.java").java """
+                package direct;
+
+                import org.junit.Assert;
+                import org.junit.Test;
+
+                public class MultiplierTest {
+                    @Test
+                    public void testMultiply() {
+                        Assert.fail("intentional failure to test jacoco coverage figures");
+                        Multiplier multiplier = new Multiplier();
```

Comment:
These lines after the intentional failure can be removed.

Reply:
Agreed